cmd/kubelet/app/options/options.go

```diff
@@ -325,7 +325,7 @@ func (f *KubeletFlags) AddFlags(mainfs *pflag.FlagSet) {
 
 	fs.StringVar(&f.HostnameOverride, "hostname-override", f.HostnameOverride, "If non-empty, will use this string as identification instead of the actual hostname. If --cloud-provider is set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used).")
 
-	fs.StringVar(&f.NodeIP, "node-ip", f.NodeIP, "IP address of the node. If set, kubelet will use this IP address for the node. If unset, kubelet will use the node's default IPv4 address, if any, or its default IPv6 address if it has no IPv4 addresses. You can pass '::' to make it prefer the default IPv6 address rather than the default IPv4 address.")
+	fs.StringVar(&f.NodeIP, "node-ip", f.NodeIP, "IP address (or comma-separated dual-stack IP addresses) of the node. If unset, kubelet will use the node's default IPv4 address, if any, or its default IPv6 address if it has no IPv4 addresses. You can pass '::' to make it prefer the default IPv6 address rather than the default IPv4 address.")
```

Comment:
the `if set` part of the doc has been removed. Is this intentional?

Reply:
It seemed like it would be awkward to rewrite the sentence in a way that reflected the fact that the value could be one or two addresses, and it seemed kind of obvious and unnecessary to state anyway... ("kubelet --node-ip tells kubelet to set the node IP")

Reply:
so, in order to have a kubelet with dual stack addresses you always have to pass always the node-ips as a parameter.
This is a pita for installation automation :/ but since you did an extensive research on this topic I imagine that is not easy to support/add it

Reply:
ic, RTFM thing , it is in the KEP